Design/ThirdpartyDependencies.md

```diff
@@ -0,0 +1,335 @@
+a gcc-7 allowing C++17-enabled builds.
+
+Pros:
+* Easier developer setup.
+* Smaller user packages. The system libraries are shared for all packages resulting
+  in a much smaller package to install.
+* (Mostly) managed for us.
+* Simple, well-known commands to install packages (apt/yum)
+
+
+Cons:
+* Dependencies not versioned with the main code. Hard to track versions used.
+* Versions become outdated very quickly. Facilities tend to use
+  older distributions that do not have up to date versions of packages.
+* Upgrading/rebuilding package versions can cause conflicts with other packages
+  not related to mantid and even old mantid versions.
+* New dependencies have to be installed manually on developers and builders machines.
+
+
+### Windows
+
+Windows uses a collection of custom-built libraries managed
+by a homegrown set of [scripts](https://github.com/mantidproject/thirdparty-msvc2015).
+The binaries are stored using the [git lfs](https://git-lfs.github.com/)
+extension for managing large binaries with git.
+For developers, all of the libraries are pulled down by
+the [cmake configure step](https://github.com/mantidproject/mantid/blob/master/buildconfig/CMake/Bootstrap.cmake#L13)
+thereby minimizing the effort required for developers to get up and running.
+It also includes a bundled version of Python as none is available by default
+on Windows.
+
+The package provided to users contains all required dependencies and is a
+self-contained bundle.
+
+Pros:
+* Easy developer setup.
+* Complete freedom to choose package versions and apply patches etc.
+* New dependencies are retrieved automatically for developers and build servers
+* Self-contained user package.
+
+Cons:
+* Dependencies not versioned with the main code. Hard to track versions used.
+* Homegrown set of scripts are very clunky.
+* Maintenance of build scripts adds overhead.
+* Complete bundling of dependencies gives a much larger user installer package.
+* ABI changes require a complete rebuild
+* Initial download (~5GB) takes quite some time
+
+
+### MacOS
+
+MacOS uses the [Homebrew](https://brew.sh/) package manager for C++ dependencies.
+A custom [tap](https://github.com/mantidproject/homebrew-mantid) provides a
+[mantid-developer](https://github.com/mantidproject/homebrew-mantid/blob/master/mantid-developer.rb)
+meta-formula to install all of the dependencies in one shot. 
+Python dependencies are installed with `pip` and a 
+[requirements file](https://github.com/mantidproject/homebrew-mantid/blob/master/requirements.txt).
+Homebrew either downloads a prebuilt binary or builds it locally from the formula.
+A developer formula is provided to easily install the required dependencies.
+
+The package provided for users contains all required dependencies
+much like Windows.
+
+Pros:
+* Formula for dependencies (mostly) maintained for us.
+* Simple commands to install packages
+
+Cons:
+* Dependencies not versioned with the main code. Hard to track versions used.
```

Comment:
I would add here that for some of the dependencies we have to maintain our own formulae, such as nexusformat, and pyqt@4 in the past.

Reply:
For the record, here we could reinstate that the bundle is over 1GB and takes long to download. Waiting few minutes for download is not an issue, if it is one time thing, unfortunately it is not, since there is no mechanism of update for nightly nor release as in linux. Each time one needs a new nightly, he needs to remove the existing, download and install the new one. Of course, the bundle size will decrease when MantidPlot and Paraview are removed, but it'll still be around 500MB.

Reply:
Just thinking further on this, would it make sense to have a brew formula for mantid itself? That way one could, in principle, upgrade in-situ on command line.

Reply:
The size is a reasonable point. The question becomes though whether a smaller binary package is worth having to stick with system libraries that can become very old.

Reply:
A brew formula for mantid would in theory be fairly easy, especially if we didn't provide bottles so this could be an option too.